File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/annotation/ExcelColumn.java
package com.jmsoftware.maf.springcloudstarter.annotation;

import org.apache.poi.ss.usermodel.CellType;

import java.lang.annotation.*;

/**
 * Description: ExcelColumn, change description here.
 *
 * @author 钟俊 （zhongjun）, email: zhongjun@toguide.cn, date: 2/19/2021 9:15 AM
 */
@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ExcelColumn {
    /**
     * Column name.
     *
     * @return the string
     */
    String name() default "Default Column Name";

    /**
     * Cell type cell type.
     *
     * @return the cell type
     */
    CellType cellType() default CellType.STRING;
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/poi/ExcelColumn.kt
package com.jmsoftware.maf.springcloudstarter.poi

import org.apache.poi.ss.usermodel.CellType

/**
 * # ExcelColumn
 *
 * Description: ExcelColumn, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 3:57 PM
 */
@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class ExcelColumn(
    /**
     * Column name.
     *
     * @return the string
     */
    val name: String = "Default Column Name",
    /**
     * Cell type.
     *
     * @return the cell type
     */
    val cellType: CellType = CellType.STRING
)
